### PR TITLE
[now-cli] Add `now build` command to allow running builds locally

### LINKED ISSUE #2614: Add `now build` command
This command will allow replicating the production behaviour 100% for testing. No testing, no development watching interface, just production.

### PR DESCRIPTION
**Pull request by ijjk**
This adds initial support for `now build` to hopefully make debugging easier. It runs the builds in the same process so we can pass custom environment variables to debug faster and we should be able to link builders in `.now/builders` to test them similar to linking them in `NOW_BUILDER_CACHE_DIR='../.now-dev'` to debug them with `now dev`

Closes #2614.

### CODE REVIEW

**Review by TooTallNate — DISMISSED** [review]

**Comment by ijjk** [comment]
Added some tests for `now build` 👍

**Comment by rauchg** [comment]
Amazing @ijjk

**Comment by styfle** [comment]
@ijjk Since this is a development feature, I think we should _not_ use `now build` as a top-level command because we may wish to use this for something else in the future. It could also be confusing for new users because they would never need this feature.

What about `now dev --export` instead? This will work better with our new debug flag because you could use `now dev --export --debug` to get verbose output from the builders (most console.log() calls will be hidden without the --debug flag in the next release).

File: packages/now-cli/src/commands/build/index.ts [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+  }
+  if (!builds || builds.length === 0) {
+    return output.warn('No builds found');
+  }
+  output.log(`Found ${plural('build', builds.length, true)}`);
+  output.debug(`builds: ${JSON.stringify(builds)}`);
+
+  output.log('Installing builders');
+  const yarnDir = await getYarnPath(output);
+  const packagesSet = new Set<string>(builds.map(build => build.use));
```
Comment (styfle) [inline packages/now-cli/src/commands/build/index.ts:118]: You need to special case `@now/static` which is not a real builder so it does not need to be installed.

File: packages/now-cli/src/commands/build/index.ts [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+const glob = async (pattern: string, options: GlobOptions) => {
+  return (await globP(pattern, options)).map((p: string) =>
+    p.replace(/\\/g, '/')
+  );
+};
+
+const cwd = process.cwd();
+const outputDir = path.join(cwd, '.now');
+const workDir = path.join(outputDir, 'workDir');
+const buildersDir = path.join(outputDir, '/builders');
```
Comment (styfle) [inline packages/now-cli/src/commands/build/index.ts]:
```suggestion
const buildersDir = path.join(outputDir, 'builders');
```

File: packages/now-cli/src/commands/build/index.ts (hunk 1 above)
Comment (ijjk) [reply packages/now-cli/src/commands/build/index.ts:118]: Added handling of `@now/static` and added a test for it 👍

File: packages/now-cli/package.json [hunk 3]
```diff
@@ -15,6 +15,7 @@
     "test-unit": "nyc ava test/*unit.js --serial --fail-fast --verbose",
     "test-integration": "ava test/integration.js --serial --fail-fast",
     "test-integration-now-dev": "ava test/dev/integration.js --serial --fail-fast --verbose",
+    "test-integration-now-build": "ava test/build/integration.js --serial --fail-fast --verbose",
```
Comment (styfle) [inline packages/now-cli/package.json:18]: I don't believe these new tests are running in CI.

You will need to configure Circle CI.

Comment (ijjk) [reply packages/now-cli/package.json:18]: Added it to CircleCi

**Review by styfle — APPROVED** [review]

**Pull request closed without merging**